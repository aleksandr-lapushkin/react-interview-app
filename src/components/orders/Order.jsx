import React from "react";
import { Orders } from "../../model";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

export default class Order extends React.PureComponent {
  render() {
    return (
      <div>

        <div className="row">
            <Link to="/">
                <button>Home</button>
            </Link>
            <button className="button-primary">Save</button>
        </div>
        
        <form>
          <div className="row u-full-width">
            <div className="columns six">
              <label htmlFor="title">Title</label>
              <input className="u-full-width" id="title" type="text" required/>
            </div>

            <div className="columns six">
              <label htmlFor="status">Status</label>
              <select className="u-full-width" id="status" required>
                {Object.keys(Orders.Status)
                    .map(status => <option key={status} value={status}>{status}</option>)}
              </select>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

Order.propTypes = {
  order: PropTypes.object
};
