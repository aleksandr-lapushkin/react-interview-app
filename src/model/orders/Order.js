export default class Order {
  constructor(id, title, status) {
    this.id = id;
    this.title = title;
    this.status = status;
  }
}
