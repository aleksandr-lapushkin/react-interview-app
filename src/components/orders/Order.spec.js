import React from "react";
import { Orders } from "../../model";
import { render } from "@testing-library/react";
import Order from "./Order";
import {StaticRouter} from "react-router";


describe("Order", () => {
  test("Snapshot matches empty form", () => {
    const result = render(<StaticRouter><Order/></StaticRouter>)
    expect(result.container).toMatchSnapshot()
  })
})
