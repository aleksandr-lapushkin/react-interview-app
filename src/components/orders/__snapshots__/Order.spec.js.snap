// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Snapshot matches empty form 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <a
        href="/"
      >
        <button>
          Home
        </button>
      </a>
      <button
        class="button-primary"
      >
        Save
      </button>
    </div>
    <form>
      <div
        class="row u-full-width"
      >
        <div
          class="columns six"
        >
          <label
            for="title"
          >
            Title
          </label>
          <input
            class="u-full-width"
            id="title"
            placeholder="Enter title..."
            required=""
            type="text"
          />
        </div>
        <div
          class="columns six"
        >
          <label
            for="status"
          >
            Status
          </label>
          <select
            class="u-full-width"
            id="status"
            required=""
          >
            <option
              value="SUBMITTED"
            >
              SUBMITTED
            </option>
            <option
              value="PROCESSING"
            >
              PROCESSING
            </option>
            <option
              value="IN_TRANSIT"
            >
              IN_TRANSIT
            </option>
            <option
              value="DELIVERED"
            >
              DELIVERED
            </option>
            <option
              value="CANCELLED"
            >
              CANCELLED
            </option>
          </select>
        </div>
      </div>
    </form>
  </div>
</div>
`;
